src/RamExpression.h

```diff
@@ -109,19 +125,10 @@ class RamIntrinsicOperator : public RamExpression {
         return new RamIntrinsicOperator(operation, std::move(argsCopy));
     }
 
-    void apply(const RamNodeMapper& map) override {
-        for (auto& arg : arguments) {
-            arg = map(std::move(arg));
-        }
-    }
-
 protected:
     /** Operation symbol */
     const FunctorOp operation;
 
-    /** Arguments of the function */
-    std::vector<std::unique_ptr<RamExpression>> arguments;
-
     bool equal(const RamNode& node) const override {
         assert(nullptr != dynamic_cast<const RamIntrinsicOperator*>(&node));
         const auto& other = static_cast<const RamIntrinsicOperator&>(node);
```

Comment:
Please call the equivalence of the superclass and add the check for operation here. 
return RamAbstractOperator::equal(other) && getOperator() == other.getOperator();